ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/components/cards-table/counts-table.component.spec.ts

```diff
@@ -0,0 +1,184 @@
+      count: 49,
+      size: 'SMALL',
+      title: 'PTs w/ Tissue Consent',
+      type: 'COUNT'
+    },
+    {
+      color: [],
+      count: 1,
+      size: 'SMALL',
+      title: 'Mr Requested',
+      type: 'COUNT'
+    }
+  ];
+
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [CountsTableComponent],
+      imports: [MatTableModule, MatSortModule, MatInputModule, MatFormFieldModule, NoopAnimationsModule]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CountsTableComponent);
+    component = fixture.debugElement.componentInstance;
+    harnessLoader = TestbedHarnessEnvironment.loader(fixture);
+
+    setTestData();
+  });
+
+  /**
+   * Only checks if component was created
+   */
+  it('should create component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  /**
+   * Checks if all data is visible by the number of rows rendered on UI
+   */
+  it('should show all data', async () => {
+    const matTableHarness: MatTableHarness = await getMatHarness<MatTableHarness>(MatTableHarness);
+    const matRowHarnesses: MatRowHarness[] = await matTableHarness.getRows();
+
+    expect(matRowHarnesses.length).toBe(3);
+  });
+
+  /**
+   * Checks if in case of no data, appropriate row is visible by count of rows (should be 1)
+   * and by its first cell's context
+   */
+  it('should show zero data', async () => {
+    component.setCounts = []; // Resetting data to an empty array
+
+    const matTableHarness: MatTableHarness = await getMatHarness<MatTableHarness>(MatTableHarness);
+    const matRowHarnesses: MatRowHarness[] = await matTableHarness.getRows();
+    const [firstCell]: MatCellHarness[] = await matRowHarnesses[0].getCells();
+    const firstCellText: string = await firstCell.getText();
+
+    expect(matRowHarnesses.length).toBe(1);
+    expect(firstCellText).toBe('No data matching the filter ""');
+  });
+
+  /**
+   * Checks after filtering by "mr" title/count, if there is displayed only one row with appropriate context
+   */
+  it('should filter one item', async () => {
+    const matRowHarnesses: MatRowHarness[] = await getTableRowsAfterFilter('mr');
+    const [firstCell]: MatCellHarness[] = await matRowHarnesses[0].getCells();
+    const firstCellText: string = await firstCell.getText();
+
+    expect(matRowHarnesses.length).toBe(1);
+    expect(firstCellText).toBe('Mr Requested');
+  });
+
+  /**
+   * Checks after filtering by non-existing title/count, if there is displayed only one row with appropriate context
+   */
+  it('should filter zero item', async () => {
+    const NOT_FOUND = 'notFound';
+    const matRowHarnesses: MatRowHarness[] = await getTableRowsAfterFilter(NOT_FOUND);
+    const [firstCell]: MatCellHarness[] = await matRowHarnesses[0].getCells();
+    const firstCellText: string = await firstCell.getText();
+
+    expect(firstCellText).toBe('No data matching the filter ' + '"' + NOT_FOUND + '"');
```

Comment:
**_#optional_**

Template literals with backticks ( ` ) can be used here as a short option.

Reply:
Oh, forgot to fix that. Fixed!